lib/transactions.js

```diff
@@ -1,44 +1,29 @@
+import { assertOkResponse } from './http-assertions.js'
 import { pickRandomNumber } from './random.js'
 
-// TODO: Find a publicly documented API to replace Viewblock API
-
 /**
- * Fetches a random list of confirmed transaction IDs from the Arweave network.
+ * Fetches a random block from the Arweave network and returns the IDs of its transactions.
  *
- * @param {Function} fetchTransactions - A function to fetch transactions from a specific page.
- * @returns {Promise<string[]>} A promise that resolves to an array of confirmed transaction IDs.
+ * @param {typeof globalThis.fetch} fetch - The fetch function to use.
+ * @returns {Promise<string[]>} A promise that resolves to an array of transaction IDs.
  */
-export const getTransactions = async (fetchTransactions = fetchTransactionsFromViewblock) => {
-  // We hardcode the min and max page numbers for now to avoid hitting the API too much
-  // Transactions before page 50 are less likely to be confirmed
-  const minPage = 50
-  // Max page set to 10,000 to fetch recent transactions. Adjust as needed (max is around 2.8M).
-  const maxPage = 10_000
-  const randomPageNumber = pickRandomNumber(minPage, maxPage)
+export const getTransactions = async (fetch = globalThis.fetch) => {
+  console.log('Fetching current network info')
+  const infoResp = await fetch('https://arweave.net/info')
+  await assertOkResponse(infoResp, 'Failed to fetch network info')
+  const info = await infoResp.json()
 
-  const res = await fetchTransactions(randomPageNumber)
-  const transactions = await res.json()
-  return transactions.docs.reduce((acc, tx) => {
-    if (tx.status === 'confirmed') {
-      acc.push(tx.id)
-    }
-    return acc
-  }, [])
-}
+  console.log(`Current network height: ${info.height}`)
 
-/**
- * Fetches transactions from the Viewblock API for the Arweave network.
- *
- * @param {number} [page=0] - The page number to fetch transactions from.
- * @returns {Promise<Response>} A promise that resolves to the fetch response.
- */
-const fetchTransactionsFromViewblock = async (page = 0) => {
-  return fetch(
-    `https://api.viewblock.io/arweave/txs?page=${page}&network=mainnet`,
-    {
-      headers: {
-        Origin: 'https://viewblock.io'
-      }
-    }
-  )
+  const minBlock = 2
+  const randomBlockNumber = pickRandomNumber(minBlock, info.height)
+
+  console.log(`Fetching block ${randomBlockNumber}`)
+
+  const randomBlockResp = await fetch(`https://arweave.net/block/height/${randomBlockNumber}`)
+  await assertOkResponse(randomBlockResp, 'Failed to fetch random block')
+  const { txs } = await randomBlockResp.json()
+
+  console.log(`Fetched block ${randomBlockNumber} with ${txs.length} transactions`)
+  return txs
```

Comment:
This happened when manually running the module:

```
Fetching block 73793
Fetched block 73793 with 0 transactions
Found 0 transactions
Retrieving transaction undefined from http://176.9.24.56:1984
Failed to retrieve transaction undefined from http://176.9.24.56:1984: Error: Unable to get transaction offset: invalid_address
```

Reply:
Great catch! I have addressed that issue  in https://github.com/CheckerNetwork/arweave-checker/pull/21/commits/479612b0b9a6605952aa8b14f04153d5a3fcdec9. 

I considered retrying the request with a different random block number if there were no transactions in the block, but I decided against it to prevent spamming the API (node).

Reply:
Sounds good to me!